src/prefect/states.py

```diff
@@ -126,12 +127,12 @@ def format_exception(exc: BaseException, tb: TracebackType = None) -> str:
             f"{exc_type.__module__}.{exc_type.__name__}", exc_type.__name__
         )
 
-    return formatted
+    # HACK(asm,2023-10-12): truncate exceptions to 100,000 characters
+    return truncated_to(100_000, formatted)
```

Comment:
Heck yeah

If you want to take this value from a prefect setting instead of hard-coding something, you could probably copy the boilerplate stuff I did in an earlier PR

Reply:
Specifically that be [this earlier commit](https://github.com/Jellyfish-AI/prefect/pull/1/commits/1915a05466a73568ef47f8704ae91010f99f9a85)

Reply:
yeah I thought a little bit about this, I basically matched what the [`_emit_task_run_state_change_event`](https://github.com/Jellyfish-AI/prefect/blob/783cb2b06bf7527887d75cb23b35d9f87b3dfe7a/src/prefect/engine.py#L2415-L2442) method was already doing instead of pulling the truncation length into a setting

if we want to push to get merged upstream we can clean this up + add some tests, but I'd like to get this merged here as-is to avoid having long exceptions happen again